pkg/controller/application/reconciliation/application_reconciler.go

```diff
@@ -16,10 +18,15 @@ const applicationReconcilerIdentifier = "kubernetes manifests"
 // applicationReconciler contains service and metadata for the relevant resource
 type applicationReconciler struct {
 	client client.ApplicationService
+	logger logging.Logger
 }
 
 // Reconcile knows how to do what is necessary to ensure the desired state is achieved
 func (a *applicationReconciler) Reconcile(ctx context.Context, meta reconciliation.Metadata, state *clientCore.StateHandlers) (reconciliation.Result, error) {
+	a.logger = logging.GetLogger("application", "reconcile")
+	a.logger.Debug(fmt.Sprintf("Reconciling application: %#v \n", meta.ApplicationDeclaration.Image))
```

Comment:
%#v will print like this
`Reconciling application: v1alpha1.ApplicationImage{Name:"okctl-reference-app", URI:""}`

Reply:
spew.Sdump will print like this
```(reconciliation.Metadata) {
 Out: (*os.File)(0xc000010018)({
  file: (*os.file)(0xc0000aa0c0)({
   pfd: (poll.FD) {
    fdmu: (poll.fdMutex) {
     state: (uint64) 0,
     rsema: (uint32) 0,
     wsema: (uint32) 0
    },
    Sysfd: (int) 1,
    pd: (poll.pollDesc) {
     runtimeCtx: (uintptr) <nil>
    },
    iovecs: (*[]syscall.Iovec)(<nil>),
    csema: (uint32) 0,
    isBlocking: (uint32) 1,
    IsStream: (bool) true,
    ZeroReadIsEOF: (bool) true,
    isFile: (bool) true
   },
   name: (string) (len=11) "/dev/stdout",
   dirinfo: (*os.dirInfo)(<nil>),
   nonblock: (bool) false,
   stdoutOrErr: (bool) true,
   appendMode: (bool) false
  })
 }),
 ClusterDeclaration: (*v1alpha1.Cluster)(0xc00082e028)(&TypeMeta{Kind:Cluster,APIVersion:okctl.io/v1alpha1,}),
 ApplicationDeclaration: (v1alpha1.Application) &TypeMeta{Kind:Application,APIVersion:okctl.io/v1alpha1,},
 Purge: (bool) false,
 Logger: (*logrus.Logger)(0xc0001c6150)({
  Out: (*os.File)(0xc000010020)({
   file: (*os.file)(0xc0000aa120)({
    pfd: (poll.FD) {
     fdmu: (poll.fdMutex) {
      state: (uint64) 0,
      rsema: (uint32) 0,
      wsema: (uint32) 0
     },
     Sysfd: (int) 2,
     pd: (poll.pollDesc) {
      runtimeCtx: (uintptr) <nil>
     },
     iovecs: (*[]syscall.Iovec)(<nil>),
     csema: (uint32) 0,
     isBlocking: (uint32) 1,
     IsStream: (bool) true,
     ZeroReadIsEOF: (bool) true,
     isFile: (bool) true
    },
    name: (string) (len=11) "/dev/stderr",
    dirinfo: (*os.dirInfo)(<nil>),
    nonblock: (bool) false,
    stdoutOrErr: (bool) true,
    appendMode: (bool) false
   })
  }),...
```

Reply:
I prefer the first one, the other one was pretty noisy

Reply:
We won't get recursive print of structs if we just print "%#v" , I was thinking we spew.dump to trace log for big things, and use regular logging for the rest.

Reply:
Sånn ser en opts print ut
```
TRAC[0002] (*client.ScaffoldApplicationOpts)(0xc00036eb00)({
 Cluster: (v1alpha1.Cluster) &TypeMeta{Kind:Cluster,APIVersion:okctl.io/v1alpha1,},
 Application: (v1alpha1.Application) &TypeMeta{Kind:Application,APIVersion:okctl.io/v1alpha1,},
 HostedZoneID: (string) (len=21) "Z05552522WYTX0NMVMUAK"
})
```

Reply:
Er ikke sikkert vi  printer hele meta objektet

Reply:
Recursive print is a good point. I just remember all the output okctl used to _spew_ out back in the day

Reply:
If it's too much to print to the console, we could set logging up so that trace level goes to file only.